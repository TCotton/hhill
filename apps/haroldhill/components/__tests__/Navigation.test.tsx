import Navigation from '../Navigation'
import { render, waitFor } from '@testing-library/react'
describe('Navigation', () => {
  const articles = [
    {
      title: 'title',
      slug: 'slug',
      pages: [
        {
          id: '1',
          title: 'page 1',
          slug: 'page-1',
          contentRichText: 'contentRichText'
        },
        {
          id: 'H1Omej68TZxpk9JFIhmWQ',
          title: 'page 2',
          slug: 'page-2',
          contentRichText: 'contentRichText'
        },
        {
          id: '3',
          title: 'page 3',
          slug: 'page-3',
          contentRichText: 'contentRichText'
        }
      ]
    }
  ]

  it('renders not return undefined', () => {
    const props = {
      articles: articles
    }
    const { baseElement } = render(<Navigation {...props} />)
    expect(baseElement).toBeDefined()
  })
  it('should create snapshot', async () => {
    const props = {
      articles: articles
    }
    const { asFragment } = await render(<Navigation {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('should create snapshot when button clicked', async () => {
    const props = {
      articles: articles
    }
    const { asFragment, findByTestId } = render(<Navigation {...props} />)
    await waitFor(async () => {
      const button = await findByTestId('button')
      button.click()
    })
    expect(asFragment()).toMatchSnapshot()
  })
  it('should include the correct number of articles', async () => {
    const props = {
      articles: articles
    }
    const { findAllByTestId, findByTestId } = await render(
      <Navigation {...props} />
    )
    await waitFor(async () => {
      const button = await findByTestId('button')
      button.click()
      expect(await findAllByTestId('article')).toHaveLength(1)
    })
  })
  it('should include the correct number of pages', async () => {
    const props = {
      articles: articles
    }
    const { findAllByTestId, findByTestId } = await render(
      <Navigation {...props} />
    )
    await waitFor(async () => {
      const button = await findByTestId('button')
      button.click()
      expect(await findAllByTestId('page')).toHaveLength(3)
    })
  })
  it('should be defined if articles is empty', () => {
    const props = {
      articles: []
    }
    const { baseElement } = render(<Navigation {...props} />)
    expect(baseElement).toBeDefined()
  })
  it('should be defined if pages is empty', () => {
    const props = {
      articles: [
        {
          title: 'title',
          slug: 'slug',
          pages: []
        }
      ]
    }

    const { baseElement } = render(<Navigation {...props} />)
    expect(baseElement).toBeDefined()
  })
  it('should be defined if pages is missing', () => {
    const props = {
      articles: [
        {
          title: 'title',
          slug: 'slug'
        }
      ]
    }

    const { baseElement } = render(<Navigation {...props} />)
    expect(baseElement).toBeDefined()
  })
})
