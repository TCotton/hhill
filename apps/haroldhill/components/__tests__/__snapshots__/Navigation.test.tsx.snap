// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should create snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Main"
    class="NavigationMenuRoot"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      style="position: relative;"
    >
      <ul
        class="NavigationMenuList"
        data-orientation="horizontal"
        dir="ltr"
      >
        <li>
          <button
            aria-controls="radix-:r2:-content-radix-:r3:"
            aria-expanded="false"
            class="NavigationMenuTrigger .gem-c-layout-super-navigation-header__navigation-top-toggle-button"
            data-radix-collection-item=""
            data-state="closed"
            data-testid="button"
            id="radix-:r2:-trigger-radix-:r3:"
          >
            Menu 
            <svg
              aria-hidden="true"
              class="CaretDown"
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4.18179 6.18181C4.35753 6.00608 4.64245 6.00608 4.81819 6.18181L7.49999 8.86362L10.1818 6.18181C10.3575 6.00608 10.6424 6.00608 10.8182 6.18181C10.9939 6.35755 10.9939 6.64247 10.8182 6.81821L7.81819 9.81821C7.73379 9.9026 7.61934 9.95001 7.49999 9.95001C7.38064 9.95001 7.26618 9.9026 7.18179 9.81821L4.18179 6.81821C4.00605 6.64247 4.00605 6.35755 4.18179 6.18181Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
    <div
      class="ViewportPosition"
    />
  </nav>
</DocumentFragment>
`;

exports[`Navigation should create snapshot when button clicked 1`] = `
<DocumentFragment>
  <nav
    aria-label="Main"
    class="NavigationMenuRoot"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      style="position: relative;"
    >
      <ul
        class="NavigationMenuList"
        data-orientation="horizontal"
        dir="ltr"
      >
        <li>
          <button
            aria-controls="radix-:r4:-content-radix-:r5:"
            aria-expanded="true"
            class="NavigationMenuTrigger .gem-c-layout-super-navigation-header__navigation-top-toggle-button"
            data-radix-collection-item=""
            data-state="open"
            data-testid="button"
            id="radix-:r4:-trigger-radix-:r5:"
          >
            Menu 
            <svg
              aria-hidden="true"
              class="CaretDown"
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4.18179 6.18181C4.35753 6.00608 4.64245 6.00608 4.81819 6.18181L7.49999 8.86362L10.1818 6.18181C10.3575 6.00608 10.6424 6.00608 10.8182 6.18181C10.9939 6.35755 10.9939 6.64247 10.8182 6.81821L7.81819 9.81821C7.73379 9.9026 7.61934 9.95001 7.49999 9.95001C7.38064 9.95001 7.26618 9.9026 7.18179 9.81821L4.18179 6.81821C4.00605 6.64247 4.00605 6.35755 4.18179 6.18181Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <span
            aria-hidden="true"
            style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            tabindex="0"
          />
          <span
            aria-owns="radix-:r4:-content-radix-:r5:"
          />
        </li>
      </ul>
    </div>
    <div
      class="ViewportPosition"
    >
      <div
        class="NavigationMenuViewport"
        data-orientation="horizontal"
        data-state="open"
      >
        <div
          aria-labelledby="radix-:r4:-trigger-radix-:r5:"
          data-orientation="horizontal"
          dir="ltr"
          id="radix-:r4:-content-radix-:r5:"
        >
          <ul
            class="govuk-list govuk-list--spaced gem-c-layout-super-navigation-header__navigation-second-items"
          >
            <span>
              <li
                class="gem-c-layout-super-navigation-header__dropdown-list-item"
                data-testid="article"
              >
                <p
                  class="govuk-heading-s"
                >
                  title
                </p>
                <ul>
                  <li
                    class="gem-c-layout-super-navigation-header__dropdown-list-item"
                    data-testid="ListItem"
                  >
                    <p
                      class="govuk-link"
                    >
                      <a
                        class="ListItemLink govuk-link gem-c-layout-super-navigation-header__navigation-second-item-link"
                        data-radix-collection-item=""
                        data-testid="page"
                        href="page-1-1"
                      >
                        page 1
                      </a>
                    </p>
                  </li>
                  <li
                    class="gem-c-layout-super-navigation-header__dropdown-list-item"
                    data-testid="ListItem"
                  >
                    <p
                      class="govuk-link"
                    >
                      <a
                        class="ListItemLink govuk-link gem-c-layout-super-navigation-header__navigation-second-item-link"
                        data-radix-collection-item=""
                        data-testid="page"
                        href="page-2-H1Omej68TZxpk9JFIhmWQ"
                      >
                        page 2
                      </a>
                    </p>
                  </li>
                  <li
                    class="gem-c-layout-super-navigation-header__dropdown-list-item"
                    data-testid="ListItem"
                  >
                    <p
                      class="govuk-link"
                    >
                      <a
                        class="ListItemLink govuk-link gem-c-layout-super-navigation-header__navigation-second-item-link"
                        data-radix-collection-item=""
                        data-testid="page"
                        href="page-3-3"
                      >
                        page 3
                      </a>
                    </p>
                  </li>
                </ul>
              </li>
            </span>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
